cura/Settings/MachineManager.py

```diff
@@ -145,13 +145,13 @@ def printerOutputDevices(self):
 
     @pyqtProperty(int, constant=True)
     def totalNumberOfSettings(self) -> int:
-        return len(ContainerRegistry.getInstance().findDefinitionContainers(id="fdmprinter")[0].getAllKeys())
+        return len(ContainerRegistry.getInstance().findDefinitionContainers(id = "fdmprinter")[0].getAllKeys())
 
     def _onHotendIdChanged(self, index: Union[str, int], hotend_id: str) -> None:
         if not self._global_container_stack:
             return
 
-        containers = ContainerRegistry.getInstance().findInstanceContainers(type="variant", definition=self._global_container_stack.getBottom().getId(), name=hotend_id)
+        containers = ContainerRegistry.getInstance().findInstanceContainersMetadata(type = "variant", definition = self._global_container_stack.getBottom().getId(), name = hotend_id)
```

Comment:
replace getBottom with definition, it makes the purpose a lot more clear

Reply:
I didn't use `getBottom`, that was already there. Anyway, I replaced all occurrences in this class of `getBottom()` with `definition`: 430f3dad27b2ad6e5d1eeddd258c045ef585fa08